bgpd/bgpd.c

```diff
@@ -5512,16 +5513,17 @@ int peer_password_set(struct peer *peer, const char *password)
 			XFREE(MTYPE_PEER_PASSWORD, member->password);
 		member->password = XSTRDUP(MTYPE_PEER_PASSWORD, password);
 
+		/* Attempt to install password on socket. */
```

Comment:
What does this change accomplish?

Reply:
when setting the password on the bgp socket, first we should apply the password on the socket and then, reset the session or send update to peer.

Reply:
in the existing code, first we were sending update on the bgp socket to the neighbor (or sending reset to neighbor). Then we were updating the same socket with MD5 parameters.